diff --git a/src/utils/Cursor.ts b/src/utils/Cursor.ts
index 0317ece77139f429079ab928046a9399e8038517..5c22ac424726fa732b9eb57af42db011e021b089 100644
--- a/src/utils/Cursor.ts
+++ b/src/utils/Cursor.ts
@@ -453,9 +453,17 @@ export class Cursor {
   }
 
   endOfLine(): Cursor {
-    const { line } = this.getPosition()
-    const column = this.measuredText.getLineLength(line)
-    const offset = this.getOffset({ line, column })
+    const { line, column } = this.getPosition()
+    const lineLength = this.measuredText.getLineLength(line)
+    if (column >= lineLength && line < this.measuredText.lineCount - 1) {
+      const nextLineLength = this.measuredText.getLineLength(line + 1)
+      const nextLineOffset = this.getOffset({
+        line: line + 1,
+        column: nextLineLength,
+      })
+      return new Cursor(this.measuredText, nextLineOffset, 0)
+    }
+    const offset = this.getOffset({ line, column: lineLength })
     return new Cursor(this.measuredText, offset, 0)
   }
 
